nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,611 @@
+            .allowableValues(buildWeakCryptoAllowableValues())
+            .defaultValue(buildDefaultWeakCryptoAllowableValue().getValue())
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("Any FlowFile that is successfully encrypted or decrypted will be routed to success").build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("Any FlowFile that cannot be encrypted or decrypted will be routed to failure").build();
+    private List<PropertyDescriptor> properties;
+
+    private Set<Relationship> relationships;
+
+    static {
+        // add BouncyCastle encryption providers
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    private static AllowableValue[] buildKeyDerivationFunctionAllowableValues() {
+        final KeyDerivationFunction[] keyDerivationFunctions = KeyDerivationFunction.values();
+        List<AllowableValue> allowableValues = new ArrayList<>(keyDerivationFunctions.length);
+        for (KeyDerivationFunction kdf : keyDerivationFunctions) {
+            allowableValues.add(new AllowableValue(kdf.name(), kdf.getName(), kdf.getDescription()));
+        }
+
+        return allowableValues.toArray(new AllowableValue[0]);
+    }
+
+    private static AllowableValue[] buildEncryptionMethodAllowableValues() {
+        final EncryptionMethod[] encryptionMethods = EncryptionMethod.values();
+        List<AllowableValue> allowableValues = new ArrayList<>(encryptionMethods.length);
+        for (EncryptionMethod em : encryptionMethods) {
+            allowableValues.add(new AllowableValue(em.name(), em.name(), em.toString()));
+        }
+
+        return allowableValues.toArray(new AllowableValue[0]);
+    }
+
+    private static AllowableValue[] buildWeakCryptoAllowableValues() {
+        List<AllowableValue> allowableValues = new ArrayList<>();
+        allowableValues.add(new AllowableValue(WEAK_CRYPTO_ALLOWED_NAME, "Allowed", "Operation will not be blocked and no alerts will be presented " +
+                "when unsafe combinations of encryption algorithms and passwords are provided"));
+        allowableValues.add(buildDefaultWeakCryptoAllowableValue());
+        return allowableValues.toArray(new AllowableValue[0]);
+    }
+
+    private static AllowableValue buildDefaultWeakCryptoAllowableValue() {
+        return new AllowableValue(WEAK_CRYPTO_NOT_ALLOWED_NAME, "Not Allowed", "When set, operation will be blocked and alerts will be presented to the user " +
+                "if unsafe combinations of encryption algorithms and passwords are provided on a JVM with limited strength crypto. To fix this, see the Admin Guide.");
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(ATTRIBUTES_TO_ENCRYPT);
```

Comment:
As this is not a required property, it should be moved below the required properties.